tests/functional/oauth1.py

```diff
@@ -0,0 +1,82 @@
+"""A collection of fixtures for making signed launches."""
+
+import random
+import time
+
+import oauthlib.common
+import oauthlib.oauth1
+import pytest
+
+from tests import factories
+
+__all__ = (
+    "do_lti_launch",
+    "oauth1_credentials",
+    "application_instance",
+    "oauth1_sign_data",
+)
+
+
```

Comment:
This stack of fixtures works differently from the other ones. In this case you call the launch fixture and it signs the params for you.

In the previous set, you have to sign them first, which is awkward because you can only sign them with the URL. This makes the existing setup much more clumsy.

It's a potential next step to try and refactor the existing tests to use this sort of arrangement.